### PR TITLE
Load UI from files

### LINKED ISSUE #833: Load UI from file
Add the ability to load UI graphs from a file.
This means less messing around in the C++ side to do UI.
The C++ will then get a `Control` back that they can query for elements that it needs.
This should work nicely with DI, so the UI elements can just have a UI source or something similar.

- [ ] Create loader from file (json)
- [ ] Add required interfaces for DI
- [ ] Load a window using this - SettingsWindow seems like a good candidate.

### PR DESCRIPTION
**Pull request by chreden**
Load UI from files instead of creating them in code.
Added a JSON schema to `trview.ui/schema.json`. This is referenced by the new UI files in `trview.app/Resources/Files/UI`.
Those files are added as resources and the UI code in `trview.app` loads them from a resource then finds the elements it needs by name. They then sets up events as before.
Added `Names` structs to UI that didn't have it before.
Added some missing tests for UI.
I will add tests for the JSON deserialization in #855 as this PR is already too big.
Closes #833

### CODE REVIEW

File: trview.app.tests/UI/ViewerUITests.cpp [hunk 1]
```diff
@@ -8,6 +8,8 @@
 #include <trview.app/Mocks/UI/ISettingsWindow.h>
 #include <trview.app/Mocks/UI/IViewOptions.h>
 #include <trview.app/Mocks/UI/IContextMenu.h>
+#include <trview.app/Mocks/UI/ICameraControls.h>
```
Comment (chreden) [inline trview.app.tests/UI/ViewerUITests.cpp:11]: Should add tests for the newly testable CameraControls

File: trview.app/Resources/trview.app.rc [hunk 2]
```diff
@@ -253,14 +253,32 @@ IDB_SETTINGS            PNG                     "Files\\settings.png"
...
 
-IDR_TEXTURE_LIST        TEXT                    "Files\\texturelist.txt"
-
-IDR_FONT_LIST           TEXT                    "Files\\fontlist.txt"
-
-IDR_TYPE_NAMES          TEXT                    "Files\\type_names.txt"
-
-IDR_ACTIONS             TEXT                    "Files\\actions.json"
-
+IDR_TEXTURE_LIST                TEXT            "Files\\texturelist.txt"
```
Comment (chreden) [inline trview.app/Resources/trview.app.rc:256]: Just squash all this whitespace

File: trview.app/Routing/Route.cpp [hunk 3]
```diff
@@ -368,7 +368,7 @@ namespace trview
         auto route = route_source();
         if (json["colour"].is_string())
         {
-            route->set_colour(named_colour(to_utf16(json["colour"].get<std::string>())));
+            route->set_colour(from_colour_code(to_utf16(json["colour"].get<std::string>())));
```
Comment (chreden) [inline trview.app/Routing/Route.cpp:371]: This needs to handle existing routes that already saved a colour as a code.

File: trview.app/UI/CameraPosition.h [hunk 4]
```diff
@@ -14,9 +14,18 @@ namespace trview
     class CameraPosition final
     {
     public:
+        struct Names
```
Comment (chreden) [inline trview.app/UI/CameraPosition.h:17]: Update tests to use Names

File: trview.app/Routing/Route.cpp (hunk 3 above)
Comment (chreden) [reply trview.app/Routing/Route.cpp:371]: This is done already, codes were called names before confusingly.

File: trview.ui/json.cpp [hunk 5]
```diff
@@ -0,0 +1,263 @@
+#include "json.h"
```
[the remark this answers is not in the document]
Comment (chreden) [reply trview.ui/json.cpp:1]: #855

**Review by makotocchi — APPROVED** [review]

**Pull request merged**